packages/composites/augur/src/dataProviders/sportsdataio.ts

```diff
@@ -119,14 +217,155 @@ export const create: Execute = async (input) => {
+  const input = {
+    id,
+    data: {
+      sport,
+      league,
+      season,
+      endpoint: 'schedule'
+    }
+  }
+  const response = await exec(input)
+  return (response.result as FightSchedule[])
+    .filter(event => event.Active)
+}
+
+const getFights = async (id: string, sport: string, eventId: number, exec: Execute): Promise<Fight[]> => {
+  const input = {
+    id,
+    data: {
+      sport,
+      eventId,
+      endpoint: 'event'
+    }
+  }
+  const response = await exec(input)
+  const fights = (response.result as FightEvent).Fights
+  return fights.filter(fight => fight.Active)
+}
+
+export const createFighter: Execute = async (input) => {
+  const validator = new Validator(input, createParams)
+  if (validator.error) throw validator.error
+
+  const sport = validator.validated.data.sport.toLowerCase()
+  if (!SPORTS_SUPPORTED.includes(sport)) {
+    throw Error(`Unknown sport for Sportsdataio: ${sport}`)
+  }
+
+  const daysInAdvance = validator.validated.data.daysInAdvance
+  // const contract: ethers.Contract = validator.validated.data.contract
+
+  const sportsdataioExec = Sportsdataio.makeExecute(Sportsdataio.makeConfig(Sportsdataio.NAME))
+
+  const fights: Fight[] = []
+
+  const leagues = await getLeagues(input.id, sport, sportsdataioExec)
+  for (const league of leagues) {
+    const schedule = (await getFightSchedule(input.id, sport, league, `${new Date().getFullYear()}`, sportsdataioExec))
+      .filter(event => event.Status === "Scheduled")
+
+    for (const event of schedule) {
+      const eventFights = (await getFights(input.id, sport, event.EventId, sportsdataioExec))
+        .filter(fight => fight.Status === "Scheduled")
+        .map(fight => ({ ...fight, DateTime: event.DateTime }))
+      fights.push(...eventFights)
+    }
+  }
+
+  Logger.debug(`Augur sportsdataio: Got ${fights.length} fights from data provider`)
+  let skipNullDate = 0, skipDaysInAdvance = 0, skipOddNumberFighters = 0, cantCreate = 0
+
+  // filter markets and build payloads for market creation
+  const createEvents: CreateFighterEvent[] = []
+  for (const fight of fights) {
+    if (!fight.DateTime) {
+      skipNullDate++
+      continue
+    }
+    const startTime = Date.parse(fight.DateTime)
+    const diffTime = startTime - Date.now()
+    if (diffTime / (1000 * 3600 * 24) > daysInAdvance) {
+      skipDaysInAdvance++
+      continue
+    }
+
+    // TODO: How to check if fight already exists?
+    const canCreate = false
+    if (!canCreate) {
+      cantCreate++
+      continue
+    }
+
+    const fighters = fight.Fighters
+      .filter(fighter => fighter.Active)
+    if (fighters.length !== 2) {
+      skipOddNumberFighters++
+      continue
+    }
+
+    createEvents.push({
+      id: BigNumber.from(fight.FightId),
+      fighterA: fighters[0].FighterId,
+      fighterB: fighters[1].FighterId,
+      startTime,
+      overUnderLine: 0, // TODO: What to put here?
```

Comment:
Since O/U isn't being done at the moment, the interface for MMALinkFactory has changed.

https://github.com/AugurProject/turbo/blob/dev/packages/smart/contracts/turbo/MMALinkMarketFactory.sol#L93
```
    function createMarket(
        uint256 _eventId,
        string memory _homeFighterName,
        uint256 _homeFighterId,
        string memory _awayFighterName,
        uint256 _awayFighterId,
        uint256 _startTimestamp,
        int256[2] memory _moneylines
        ```

Reply:
Should fighter name be both first and last name?
The fighters are given as an array from the data provider, so I assume order cannot be guaranteed. This should still be fine, right? (looks like the resolve function takes both ids anyways, so doesn't seem like a problem, but would be great to confirm)

Reply:
Fighter name should be the full name (first + last).

Re: Home vs. Away -- the actual important thing here is that the fighterID lines up with the odds -- this home/away nomenclature is how TheRundown does it. As long as things are consistent I believe this works just fine.